src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -136,13 +139,33 @@ public TopicRepository getTopicRepository(final EventTypeBase eventType)
     }
 
     public TopicRepository getDefaultTopicRepository() throws TopicRepositoryException {
+        final Storage storage = getCachedDefaultStorage();
+        return topicRepositoryHolder.getTopicRepository(storage);
+    }
+
+    private Storage getCachedDefaultStorage() throws TimelineException {
+        readWriteLock.readLock().lock();
+        try {
+            if (defaultStorage != null) {
+                return defaultStorage;
+            }
+        } finally {
+            readWriteLock.readLock().unlock();
+        }
+
+        readWriteLock.writeLock().lock();
```

Comment:
@v-stepanov the lock is needed to properly initialise the storage, but I will change this impl, because there is simpler way to do it.